blackbox/docs/sql/ddl.txt

```diff
@@ -1049,3 +1049,244 @@ means that if you re-create the table using the resulting ``CREATE TABLE``
+     +--------------------+
+     | multiply(1, NULL)  |
+     +--------------------+
+     | NULL               |
+     +--------------------+
+     SELECT 1 rows in set (... sec)
+
+
+``RETURNS NULL ON NULL INPUT`` and ``STRICT``
+.............................................
+
+The ``RETURNS NULL ON NULL INPUT`` indicates that the function always returns
+``NULL`` when any of it's arguments are ``NULL``. The ``STRICT`` is a synonym
+and behaves in exactly the same way.
+
+Here's how to use ``RETURNS NULL ON NULL INPUT``::
+
+     cr> CREATE OR REPLACE FUNCTION divide(integer, integer)
+     ...  RETURNS float
+     ...  LANGUAGE JAVASCRIPT
+     ...  RETURNS NULL ON NULL INPUT
+     ...  AS 'function(a, b){
+     ...        return a / b;
+     ...      }';
+
+     cr> SELECT divide(1, NULL);
+     +--------------------+
+     | multiply(1, NULL)  |
+     +--------------------+
+     | NULL               |
+     +--------------------+
+     SELECT 1 rows in set (... sec)
+
+Here's how to use ``STRICT``::
+
+     cr> CREATE OR REPLACE FUNCTION divide(integer, integer)
+     ...  RETURNS float
+     ...  LANGUAGE JAVASCRIPT
+     ...  STRICT
+     ...  AS 'function(a, b){
+     ...        return a / b;
+     ...      }';
+
+     cr> SELECT divide(1, NULL);
+     +--------------------+
+     | multiply(1, NULL)  |
+     +--------------------+
+     | NULL               |
+     +--------------------+
+     SELECT 1 rows in set (... sec)
+
+OVERLOADING
+-----------
+
+Within a specific schema, you can overload functions by defining two functions with the
+same name that have a different set of arguments
+
+This would overload our ``add`` function with different argument types::
+
+     cr> CREATE FUNCTION my_schema.add(long, long)
+     ...  RETURNS long
+     ...  LANGUAGE JAVASCRIPT
+     ...  AS 'function(a, b){
+     ...        return a + b;
+     ...      }';
+
+This would overload our ``add`` function with more arguments::
+
+     cr> CREATE FUNCTION my_schema.add(long, long, long)
+     ...  RETURNS long
+     ...  LANGUAGE JAVASCRIPT
+     ...  AS 'function(a, b, c){
+     ...        return a + b + c;
+     ...      }';
+
+This would overload our ``add`` function with different argument names::
+
+     cr> CREATE FUNCTION my_schema.add(argument_1 long, argument_2 long)
+     ...  RETURNS long
+     ...  LANGUAGE JAVASCRIPT
+     ...  AS 'function(a, b){
+     ...        return a + b;
+     ...      }';
+
+This would overload our ``add`` function with a different return type::
+
+     cr> CREATE FUNCTION my_schema.add(long, long)
+     ...  RETURNS long
+     ...  LANGUAGE JAVASCRIPT
+     ...  AS 'function(a, b){
+     ...        return a + b;
+     ...      }';
+
+Determinism
+-----------
+
+User-defined functions are deterministic, meaning that they will always return
+the same result value when called with the same argument values. These return
+values are then cached, preventing multiple function executions.
```

Comment:
These return values COULD be cached to prevent multiple function executions, so we don't guarantee that the function is executed for each row if the argument values are the same.

IMHO we want to point out that he must not rely on the fact that this is cached / that his function is executed for each row.